src/app/products/pages/product-list-page/product-list-page.component.ts

```diff
@@ -9,18 +11,57 @@ import {Product} from "../../models/product";
 })
 export class ProductListPageComponent implements OnInit {
 
-  title = 'Tous les produits';
+  title = '';
+  count = 400;
   products: Product[] = [];
+  products$!: Observable<Product[]>;
+
+  pagesTitles = ['Meilleures ventes', 'Page de test'];
+  fullRoutes = ['/products/bestsellers', '/products/test'];
+  routes = ['bestsellers', 'test'];
 
   constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
```

Comment:
Si tu crées un objet RouteAssociation ex: 
```ts
{
   title: 'Meilleures ventes',
   func: (): Observable<Product> => this.productsService.getBestsellers(this.count)
}
```

que tu le mets dans une map -> `routeMap = new Map<string, RouteAssociation>();` puis : 
```ts
routeMap.push({
   title: 'Meilleures ventes',
   func: (): Observable<Product> => this.productsService.getBestsellers(this.count)
});
```

dans ta fonction routeSwitch tu as juste à vérifier `this.routeMap.has(activeRoute)`